numba/core/typing/templates.py

```diff
@@ -402,6 +402,15 @@ class CallableTemplate(FunctionTemplate):
     def apply(self, args, kws):
         generic = getattr(self, "generic")
         typer = generic()
+        match_sig = inspect.signature(typer)
+        try:
+            match_sig.bind(*args, **kws)
+        except TypeError as e:
+            # ctor or bind failed, raise, if there's a
```

Comment:
I think we only catch a type error in the bind now with recent changes?

Reply:
Agree, will fix.